### ISSUE TITLE
Focus is lost when showing Teaching Bubble with forceFocusInsideTrap = false

### ISSUE THREAD

**Issue by hoovercj** [microsoft/fluentui#17026]
### Environment Information

- **Package version(s)**: 7.160
- **Browser and OS versions**: Edge/Chrome 88 + Win 10

### Describe the issue:

When showing a Teaching Bubble with `focusTrapZoneProps={{ forceFocusInsideTrap: false }}` and using `calloutProps={{ hidden: !teachingBubbleVisible }}` to improve rendering performance as is recommended in the Callout component docs, focus is set to the Teaching Bubble dialog element instead of the interactive elements inside.

This seems to occur because even though the focusTrapZone does set focus to an action button, [the callout `hidden` prop updating](https://github.com/microsoft/fluentui/blob/master/packages/react/src/components/Callout/CalloutContent.base.tsx#L274) also triggers the callout component to set focus to the first focusable element which in the `TeachingBubble` is the [dialog element](https://github.com/microsoft/fluentui/blob/master/packages/react/src/components/TeachingBubble/TeachingBubbleContent.base.tsx#L156).

### Please provide a reproduction of the issue in a codepen:

https://codepen.io/hoovercj/pen/QWGpEKQ

#### Actual behavior:

When using "enter" to invoke the button in the codepen, there is no visible keyboard focus because it is set to the dialog element itself and not the interactive buttons.

#### Expected behavior:

When using "enter" to invoke the button in the codepen, focus is moved to the first focusable button in the focus zone (or the element specified by `firstFocusableSelector`)

**Comment by gouttierre** [comment]
@hoovercj : thanks for filing this item with us. 

@khmakoto -  Seems you must TAB key first, and then use the ENTER button to invoke the button.  Could you confirm if this is a regression, or if this behavior is an issue?

**Comment by hoovercj** [comment]
@gouttierre @khmakoto I don't think this is a regression, I think this bug has been around since the component was introduced and is a consequence of multiple components within this composite component each trying to influence focus.

I also want to emphasize that while this does have accessibility implications, this is at its heart **not** an accessibility bug. The problem is that we are using an api to control where initial focus is set when the dialog is set using `firstFocusableSelector` but the `Callout` component is stealing the focus away meaning that it is not possible to control where initial focus goes when toggling visibility via `calloutProps={{ hidden: !teachingBubbleVisible }}` as is recommended in the docs.

**Comment by Alex1100** [comment]
Try making the clickEvent only fire off when the eventPhase === 3

This will ensure that unless you explicitly close out the coachmark via keyboard or a mouse click, it won't lose focus no matter what else tries to steal it. :)

**Comment by hoovercj** [comment]
@Alex1100 I'm not sure I understand your comment -- I am explicitly setting `forceFocusInsideTrap: false` because I don't want the focus trap to keep focus inside the Teaching Bubble (not coachmark).

If we are conditionally rendering the callout, this works fine. Something like this:

```tsx
props.renderTeachingBubble && <TeachingBubble
    focusTrapZoneProps={{
        forceFocusInsideTrap: false,
    }}
/>
```

The problem occurs when we follow the advice in the callout component documentation which says it is better to use the `hidden` property like this:

```tsx
<TeachingBubble
    calloutProps={{
        hidden: !props.renderTeachingBubble
    }}
    focusTrapZoneProps={{
        forceFocusInsideTrap: false,
    }}
/>
```

This causes a problem because even though the focusTrapZone component correctly sets focus to an action button inside the callout, the callout component reacts to the hidden prop updating **AFTER** and sets focus to the first focusable element inside it which in the TeachingBubble is the dialog element.

One potential fix might be to update [Callout.useAutoFocus](https://github.com/microsoft/fluentui/blob/master/packages/react/src/components/Callout/CalloutContent.base.tsx#L274) so that it only focuses the first child if the callout doesn't already have focus (I haven't tested this, just reasoning about it looking at the code):

```diff
function useAutoFocus(
  { hidden, setInitialFocus }: ICalloutProps,
  positions: ICalloutPositionedInfo | undefined,
  calloutElement: React.RefObject<HTMLDivElement>,
) {
  const async = useAsync();
  const hasPositions = !!positions;
  React.useEffect(() => {
    if (!hidden && setInitialFocus && hasPositions && calloutElement.current) {
      const timerId = async.requestAnimationFrame(
-       () => focusFirstChild(calloutElement.current!),
+       () => !calloutElement.current?.contains(document.activeElement) && focusFirstChild(calloutElement.current!)
        calloutElement.current,
      );

      return () => async.cancelAnimationFrame(timerId);
    }
  }, [hidden, hasPositions, async, calloutElement, setInitialFocus]);
}
```
